### ISSUE TITLE
Should query type be string or array?

### ISSUE THREAD

**Issue by christianbundy** [dominictarr/ssb-query#15]
```javascript
var Query = {
  type: 'JSONString',
  description: 'the query it self. It should be the format of a map-filter-reduce query',
  optional: false
}
```

I don't think that this is right, because queries are meant to look like:

```
[{
  "$filter": {
    "foo": "bar"
  }
}]
```

This is an array of objects, not a string.

**Comment by dominictarr** [comment]
that's correct, but if it's a string, it tries to json parse it. otherwise it's quite awkward to write a query from the cli... I guess it should be JSON string or object. the pattern you use in https://github.com/dominictarr/map-filter-reduce/issues/6 works but only for very simple queries
